src/index.template.html

```diff
@@ -127,54 +170,72 @@
-    htmlWebpackPlugin.files.legacyScripts.length ) { %>
-    <script nomodule type="text/javascript">
-      // workaround for Safari 10.1 supporting module but ignoring nomodule
-      // From https://gist.github.com/samthor/64b114e4a4f539915a95b91ffd340acc
-      (function () {
-        var d = document;
-        var c = d.createElement('script');
-        if (!('noModule' in c) && 'onbeforeload' in c) {
-          var s = false;
-          d.addEventListener(
-            'beforeload',
-            function (e) {
-              if (e.target === c) {
-                s = true;
-              } else if (!e.target.hasAttribute('nomodule') || !s) {
-                return;
-              }
-              e.preventDefault();
-            },
-            true
-          );
+      htmlWebpackPlugin.files.moduleScripts &&
+      htmlWebpackPlugin.files.moduleScripts.length &&
+      htmlWebpackPlugin.files.legacyScripts &&
+      htmlWebpackPlugin.files.legacyScripts.length ) { %>
+        <script nomodule type="text/javascript">
+          // workaround for Safari 10.1 supporting module but ignoring nomodule
+          // From https://gist.github.com/samthor/64b114e4a4f539915a95b91ffd340acc
+          (function () {
+            var d = document;
+            var c = d.createElement('script');
+            if (!('noModule' in c) && 'onbeforeload' in c) {
+              var s = false;
+              d.addEventListener(
+                'beforeload',
+                function (e) {
+                  if (e.target === c) {
+                    s = true;
+                  } else if (!e.target.hasAttribute('nomodule') || !s) {
+                    return;
+                  }
+                  e.preventDefault();
+                },
+                true
+              );
 
-          c.type = 'module';
-          c.src = '.';
-          d.head.appendChild(c);
-          c.remove();
+              c.type = 'module';
+              c.src = '.';
+              d.head.appendChild(c);
+              c.remove();
+            }
+          })();
+        </script>
+      <% 
         }
-      })();
-    </script>
-    <% } %> <% for (const file of htmlWebpackPlugin.files.moduleScripts || []) {
-    %>
-    <script type="module" src="<%= file %>"></script>
-    <% } %> <% for (const file of htmlWebpackPlugin.files.legacyScripts || []) {
-    %>
-    <script nomodule defer type="text/javascript" src="<%= file %>"></script>
-    <% } %>
-    <% } %>
+        // Adding js files for module entry point
+        for (const file of htmlWebpackPlugin.files.moduleScripts || []) {
+          if (file.indexOf(htmlWebpackPlugin.entry)!==-1){
+            %> <script async type="module" src="<%= file %>"></script> <%   
+          } 
+          if (file.indexOf('.hydrate.')===-1 && file.indexOf('.index.')===-1){
+            %> <script async type="module" src="<%= file %>"></script> <% 
+          } 
+        }
+        // Adding js files for legacy entry point
```

Comment:
Unrelated to the PR itself, but I'd be interested in the proportion of users using the legacy modules instead of the more "modern" bundle if you have this number somewhere

Reply:
I don't know an easy way to get that number exactly. But I did a quick calculation of the browser usage for the month of February in analytics, and cross referencing it with `caniuse` info on which version support esmodules, gave an approximate number of 2.5% of people using InterPro that is still using legacy browsers.
I'm guessing this fluctuates a lot, for example, in the data I was looking, there was a random spike on firefox 44, that justified almost half the people using the legacy bundle.